lib/people_controller.rb

```diff
@@ -0,0 +1,73 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'list_manager'
+
+class PeopleController < ListManager
+  attr_accessor :people
+
+  def initialize
+    super
+    @people = []
+  end
+
+  def list_all_people
+    check_empty_list(@people, 'people')
+
+    @people.each_with_index do |person, index|
+      puts "#{index + 1}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    input = gets.chomp
+
+    case input
+    when '1'
+      new_student = create_student
+      @people << new_student
+      puts 'New person (Student) created successfully'
+    when '2'
+      new_teacher = create_teacher
+      @people << new_teacher
+      puts 'New person (Teacher) created successfully'
+    else
+      puts 'Please enter a valid number (1 or 2)'
+      create_person
+    end
+  end
+
+  def create_student
+    age, name = student_data
+    Student.new(age, nil, name)
+  end
+
+  def student_data
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    [age, name]
+  end
+
+  def create_teacher
+    age, name, specialization = teacher_data
+    Teacher.new(age, specialization, name)
+  end
+
+  def teacher_data
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    [age, name, specialization]
+  end
```

Comment:
- [ ] Great job working so far and the functionality is working fine ✨. but I think it would be great if you just kept the functionality of create student and create teacher since this is considered a Bad example of a Single responsibility rule and a Bad example of YAGNI and KISS as the contents of this method could've been placed somewhere else so kindly read more about [Best practices and Solid lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)  🔥💪🏻✨🤩

- You can check how the [refactor](https://github.com/Reem-lab/School-library-ruby/pull/5/files#diff-df46c71dda0c06fcf280fa5a3c58b7f74f08c2c83d8c4dd07632310ae9b700a7) is 


- [ ] I think you could refactor the functionality inside these methods like the inputs to make it better and not redundant ✨🙏🏻

![image](https://github.com/akezeth/OOP-School-Library/assets/58553711/1e7a647a-4006-4e70-a364-b94790b03bf8)